src/commcare_cloud/ansible/roles/postgresql/tasks/set_up_dbs.yml

```diff
@@ -0,0 +1,67 @@
+- name: Create PostgreSQL users
+  become: yes
+  become_user: "{{ db_is_remote and 'ansible' or 'postgres' }}"
```

Comment:
how does this work for AWS? Where do these tasks get run?

Reply:
Is that an actual RDS machine or just a VM we keep around for running tasks against RDS instances? Just wondering if there's a reason we can't just run them from the control machine (local)?

I was thinking we could standardize to run the same workflow on all our PG nodes but I think the issue is that when we install PG we don't get a root user which we can perform operations with. Is that right?

Reply:
In option (B) above, where `db_is_remote = true`, this gets run _from_ `django_manage` (see https://github.com/dimagi/commcare-cloud/pull/2114/files#diff-8a7f836d9437d4c444262eb4453d6971R28), _"against"_ an RDS instance—"against" in the sense that `login_host` in the tasks below gets set to the DNS address of the RDS instance.

> Just wondering if there's a reason we can't just run them from the control machine (local)

this was my first thought too I think back when I was first playing around with this, but I found it tricky to get that to work—it wasn't accepting local as an option or something. Possibly related to how we add `--limit=!ansible_skip` to every command, which I think silently also excludes local for some reason. My memory's hazy so this may not be the exact right story.

> I was thinking we could standardize to run the same workflow on all our PG nodes but I think the issue is that when we install PG we don't get a root user which we can perform operations with. Is that right?

Yeah that's exactly right. I though the same thing and went through the experiment of trying to get this to _always_ run remotely, to standardize the workflow, but then realized that the instances we create using this ansible role don't have a root user, and creating one before running this role would at least partially defeat the purpose, since this file is all about creating the right user set anyway. So I settled on this way of having different auth styles—using privileged linux user for a non-remote db, and using root user/password for a remote db.